### PR TITLE
Improve performance of filling long value with a single byte

### PR DESCRIPTION
**Pull request by skrzypo987**
SImple benchmarks show 600/s -> 1300/s

### CODE REVIEW

**Review by raunaqmorarka — APPROVED** [review]
Nice 👍

**Review by sopel39 — APPROVED** [review]
Do you have any numbers?

**Comment by sopel39** [comment]
cc @martint @electrum

**Comment by martint** [comment]
Have you seen any cases where this matters? This method is called one per invocation to `Slice.fill()`, and the bulk of the time spent in that method is replicating the long across the whole buffer. I suspect it will only have an impact when making lots of calls to `Slice.fill()` on very small slice objects.

**Review by dain — APPROVED** [review]

File: src/main/java/io/airlift/slice/Slice.java [hunk 1]
```diff
@@ -1302,22 +1302,18 @@ private void checkIndexLength(int index, int length)
...
     {
-        return (value & 0xFFL) << 56
-                | (value & 0xFFL) << 48
-                | (value & 0xFFL) << 40
-                | (value & 0xFFL) << 32
-                | (value & 0xFFL) << 24
-                | (value & 0xFFL) << 16
-                | (value & 0xFFL) << 8
-                | (value & 0xFFL);
+        long longValue = ((value & 0xFF) << 8) | (value & 0xFF);
```
[the remark this answers is not in the document]
Comment (dain) [reply src/main/java/io/airlift/slice/Slice.java:1307]: @sonatype-lift ignore

**Review by dain — COMMENTED** [review]
This is better code and I think pretty simple, so I'm going to merge it

**Comment by skrzypo987** [comment]
> Have you seen any cases where this matters? This method is called once per invocation to `Slice.fill()`, and the bulk of the time spent in that method is replicating the long across the whole buffer. I suspect it will only have an impact when making lots of calls to `Slice.fill()` on very small slice objects.

I observed an actual gain with slices of size ~100 bytes. The difference is small but measurable. I used only slices with sizes that are a multiple of 8 so I guess either JIT or the CPU branch prediction got rid of the second loop in the `fill` method making this matter slightly more.

**Pull request merged**